lib/db/mysql.js

```diff
@@ -566,21 +566,31 @@ module.exports = function (log, error) {
 
   // Internal
 
-  MySql.prototype.singleQuery = function (poolName, sql, params) {
+  MySql.prototype.singleQuery = function (poolName, sql, params, readUncommitted) {
     return this.getConnection(poolName)
       .then(
         function (connection) {
-          return query(connection, sql, params)
-            .then(
-              function (result) {
-                connection.release()
-                return result
-              },
-              function (err) {
-                connection.release()
-                throw err
-              }
-            )
+          if (readUncommitted) {
+            return query(connection, 'SET TRANSACTION ISOLATION LEVEL READ UNCOMMITTED')
```

Comment:
I'm pretty confused about this isolation level stuff now, sorry.

Previously, with `P.all()`, I was using `SET TRANSACTION ISOLATION LEVEL...` before each query, in the belief that it would revert to the session default after that query finished. Now, with `P.each()`, I want to top-and-tail the entire sequence so that the isolation level applies to all queries.

So, does that mean I have to use `SET SESSION TRANSACTION ISOLATION LEVEL...` to prevent it reverting to the session after each query, or did I misunderstand the docs and that is what `SET TRANSACTION ISOLATION LEVEL...` does anyway?

The top-and-tail approach is also a little bit pernickety in so much as the reset to the `READ COMMITTED` isolation level must be run in the case where an error occurs, i.e. like a `try...finally` clause. It's not a massive deal, but it's logic that could otherwise be avoided if it were possible to simply use `SET TRANSACTION ISOLATION LEVEL...` before each query and have the isolation revert to the session default automatically.

Hence, I just wanted to check what the correct interpretation of the docs should be.